pkg/controller/statefulset/stateful_controller_test.go

```diff
@@ -0,0 +1,361 @@
+package statefulset
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/pusher/wave/pkg/core"
+	"github.com/pusher/wave/test/utils"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+var _ = Describe("StatefulSet controller Suite", func() {
+	var c client.Client
+	var m utils.Matcher
+
+	var sts *appsv1.StatefulSet
```

Comment:
I'd prefer if this variable were called `statefulset` rather than `sts`, then it is consistent with the deployment controller 😃